doc/dbus_api.md

```diff
@@ -285,6 +285,113 @@ when the question is answered.
 
 - Password -> string (rw)
   Password provided to decrypt a LUKS device.
+
 - Attempt -> unsigned 32-bit integer (r)
   Current attempt to decrypt the device. This value is useful for clients to know if the very same
   question is asked again (i.e., when the provided password did not work).
+
+
+## ServiceStatus
+
+Each service will have an status (*idle* or *busy*). The service should change its status to *busy*
+when it is going to start an expensive tasks. The status should be set back to *idle* once the long
+task is done.
+
+The main object of a service implements the following interface:
+
+### org.opensuse.DInstaller.ServiceStatus1
+
+#### Properties
+
+- All -> array(array(dict(string, variant))) (r)
+
+  All possible statuses:
+  ~~~
+  [
+    {"id" => 0, "label" => "idle"},
+    {"id" => 1, "label" => "busy"}
+  ]
+  ~~~
+
+- Current -> unsigned 32-bit integer (r)
+
+  Id of the current status.
+
+
+## Progress
+
+The main object of a service implements the following interface:
+
+### org.opensuse.DInstaller.Progress1
+
+- TotalSteps: unsigned 32-bit integer (r)
+  Number of steps.
+
+- CurrentStep: struct(unsigned 32-bit integer, string) (r)
+  Number of the current step and its description.
+
+- Finished: b (r)
+  Whether the progress has finished.
+
+## Manager
+
+### Installation Phases
+
+The installation process follows a set of phases. Only the main service (`DInstaller::Manager`)
+knows the information about the current installation phase. The rest of services will act as utility
+services without any knowledge about the whole installation process.
+
+A client (e.g., a web UI) will ask to the main service for the current phase of the installation.
+
+In principle, the installation will follow 3 possible phases: *Startup*, *Config* and *Install*.
+
+* *Startup* Phase
+
+This is the initial phase. The manager service will start in this phase and it will not change to
+another phase until the client asks for performing the next phase.
+
+* *Config* Phase
+
+The installation is configured during this phase. Configuring the installation means that everything
+needed from the system is read and the required default proposal are calculated. In YaST terms, the
+*config* phase implies to probe some modules like storage, language, etc, and to perform their
+proposals. Note that not all modules have to be probed/proposed. Probing some modules could be
+delayed to the next *install* phase.
```

Comment:
Could probing languages/network happen during the Startup phase? just to clarify.

Reply:
Yes, sure. We can move to the startup phase everything we consider. The idea with the config phase is that you could "click" on install after the manager has done the config phase, that is, the manager has finished to configure/calculate everything that could be modified by a client.